url/src/lib.rs

```diff
@@ -194,7 +194,7 @@ pub struct Url {
     ///   authority = "//" userinfo? host [ ":" port ]?
     ///   userinfo = username [ ":" password ]? "@"
     ///   hierarchical-path = [ "/" path-segment ]+
-    serialization: String,
+    serialization: Cow<'static, str>,
```

Comment:
Could replace this with [`beef::lean::Cow`](https://docs.rs/beef/latest/beef/lean/type.Cow.html) to reduce size of `Url`, but might be a pre-mature optimization